Scripts/extras/pull_and_build_from_git.py

```diff
@@ -571,8 +583,28 @@ def sync_source(self):
         if self.package_info.depends_on_packages:
             for package_name, package_hash, _ in self.package_info.depends_on_packages:
                 temp_packages_folder = self.base_temp_folder
-                if not PackageDownloader.DownloadAndUnpackPackage(package_name, package_hash, str(temp_packages_folder)):
-                    raise BuildError(f"Failed to download a required dependency: {package_name}")
+                if PackageDownloader.ValidateUnpackedPackage(package_name, package_hash, str(temp_packages_folder)):
+                    print(f"Package {package_name} already downloaded")
+                else:
+                    if not PackageDownloader.DownloadAndUnpackPackage(package_name, package_hash, str(temp_packages_folder)):
+                        raise BuildError(f"Failed to download a required dependency: {package_name}")
+
+        # Check if there are any additional package dependencies to download and extract
+        if self.package_info.additional_download_packages:
+            print("Downloading additional packages")
+            for package_url, package_hash, package_algorithm in self.package_info.additional_download_packages:
+                print(f"Retrieving additional package {os.path.basename(package_url)} from {package_url}")
```

Comment:
Seems like using basename on a url will only work if after the last slash the package name is available.

So for a link of something like `http://example.org/cdn/assimp-5.0.1-rev11-multiplatform.tar.xz` it would work, but it wouldn't work for some URL that points to a file like `http://example.org/cdn/RandomHash/ef67aa11204935`

Reply:
Not sure if I follow, but the base name is purely informational, and is not really needed. I'll change to just printing the full package url instead